### PR TITLE
Autotiler docs and cleanup

### PR DESCRIPTION
**Pull request by LeoDog896**
Adds lots of docs, and cleans up the messy switch-like if statement code.

### CODE REVIEW

File: core/src/mindustry/world/blocks/Autotiler.java [hunk 1]
```diff
@@ -112,24 +144,47 @@ default TextureRegion botHalf(TextureRegion input){
...
-            bits[0] = 2;
-        }else if(num == 3){
-            bits[0] = 2;
-            bits[2] = -1;
-        }else if(num == 4){
-            bits[0] = 1;
-            bits[2] = -1;
-        }else if(num == 5){
-            bits[0] = 1;
+        switch (num){
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/Autotiler.java:154]: I don't understand why people think `switch` is somehow less messy. This is *more lines of code*, and you need a `break` statement after every case. It's not any better than an `if`-chain.

File: core/src/mindustry/world/blocks/Autotiler.java [hunk 2]
```diff
@@ -14,27 +14,53 @@
...
-     * mode == 0 -> no slice
-     * mode == 1 -> bottom
-     * mode == 2 -> top */
-    default TextureRegion sliced(TextureRegion input, int mode){
-        return mode == 0 ? input : mode == 1 ? botHalf(input) : topHalf(input);
+    /**
+     * The mode to slice a texture at.
+     */
+    enum SliceMode{
+        NO_SLICE,
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/Autotiler.java:29]: Never use `SCREAMING_SNAKE_CASE`, see [the guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#follow-the-formatting-guidelines) for why.

**Review by Anuken — CHANGES_REQUESTED** [review]

File: core/src/mindustry/world/blocks/Autotiler.java (hunk 1 above)
Comment (LeoDog896) [reply core/src/mindustry/world/blocks/Autotiler.java:154]: Really, it just matters as how its implemented throughout the code, and, after checking, this is how its used a lot of the time, so I'll revert it.

File: core/src/mindustry/world/blocks/Autotiler.java (hunk 2 above)
Comment (LeoDog896) [reply core/src/mindustry/world/blocks/Autotiler.java:29]: Ok, ill fix that now

**Pull request merged**